wakuv2/waku.go

```diff
@@ -1012,11 +1012,22 @@ func (w *Waku) publishEnvelope(envelope *protocol.Envelope, publishFn publishFn,
 	defer w.wg.Done()
 
 	var event common.EventType
-	if err := publishFn(envelope, logger); err != nil {
-		logger.Error("could not send message", zap.Error(err))
-		event = common.EventEnvelopeExpired
-	} else {
-		event = common.EventEnvelopeSent
+	for {
+		err := publishFn(envelope, logger)
```

Comment:
Is the intention here to keep trying to send the message forever in case the error received is `not enough peers to publish`? I wonder if such code shouldn't be responsibility of https://github.com/status-im/status-go/blob/0db114f54470409f541500f54c96421e4aeffd67/protocol/transport/envelopes_monitor.go#L287-L317 ?

Reply:
Also, what do you think instead of adding a `sleep` of 3 seconds, to introduce a backoff mechanism?